assets/js/base/utils/errors.js

```diff
@@ -28,3 +33,27 @@ export const formatError = async ( error ) => {
 		type: error.type || 'general',
 	};
 };
+
+/**
+ * Given an API response object, formats the error message into something more human readable.
+ *
+ * @param  {Object}   response           Response object.
+ * @return {string}   Error message.
+ */
+export const formatStoreApiErrorMessage = ( response ) => {
+	if ( response.data && response.code === 'rest_invalid_param' ) {
+		const invalidParams = Object.values( response.data.params );
+		if ( invalidParams[ 0 ] ) {
+			return invalidParams[ 0 ];
+		}
+	}
+
+	if ( response?.message ) {
+		return response?.message;
+	}
+
+	return __(
+		'Something went wrong. Please contact us to get assistance.',
+		'woo-gutenberg-products-block'
+	);
```

Comment:
This could probably be simplified to just be something like:

```suggestion
	return response?.message || __(
		'Something went wrong. Please contact us to get assistance.',
		'woo-gutenberg-products-block'
	);
```